internal/server/handler_test.go

```diff
@@ -142,6 +142,77 @@ func TestHandler(t *testing.T) {
 		// Verify response headers
 		assert.Equal(t, http.StatusNoContent, resp.StatusCode, "unexpected status code")
 	})
+
+	t.Run("escaped characters", func(t *testing.T) {
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// We test that the bucket name is stripped.
+			assert.Equal(t, "/foo bar+", r.URL.Path)
+
+			// We verify the unescaped path
+			assert.Equal(t, "/foo%20bar%2B", r.RequestURI)
```

Comment:
Confirmed this test fails without these changes, with the `RequestURI` being `/foo%20bar+` as described in the details.